### ISSUE TITLE
dplyr 1.0.8, rlang 1.0

### ISSUE THREAD

**Issue by romainfrancois** [egnha/nofrills#54]
We're about to release dplyr 1.0.8 and as part of running our rev dep checks, it seems there is an issue with this package. 

````
── After ─────────────────────────────────────────────────────────────────────────────────────────────────────
> checking examples ... ERROR
  Running examples in ‘nofrills-Ex.R’ failed
  The error most likely occurred in:
  
  > ### Name: as_fn
  > ### Title: Abbreviated functional arguments
  > ### Aliases: as_fn
  > 
  > ### ** Examples
  > 
  > call_fn <- function(.f, x) {
  +   f <- as_fn(.f)
  +   f(x)
  + }
  > call_fn(log, 1)
  [1] 0
  > call_fn(.(. ~ sin(.) ^ 2), 1)
  [1] 0.7080734
  > # simplified function expressions support quasiquotation
  > f <- sin
  > call_fn(.(. ~ (!!f)(.) ^ 2), 1)
  Error in `function_()`: Body must be an expression or closure.
  Backtrace:
      ▆
   1. ├─global call_fn(.(. ~ (!!f)(.)^2), 1)
   2. │ └─nofrills::as_fn(.f)
   3. │   └─nofrills:::interpret_fn(x, match.fun(.f), parent.frame(2))
   4. │     └─base::eval(x, env)
   5. │       └─base::eval(x, env)
   6. └─nofrills `<fn>`(. ~ (!!f)(.)^2)
   7.   └─nofrills function_(d$args, d$body, ..env)
  Execution halted

> checking tests ... ERROR
  See below...

── Test failures ─────────────────────────────────────────────────────────────────────────────── testthat ────

> library(testthat)
> library(nofrills)
> 
> test_check("nofrills")
══ Failed tests ════════════════════════════════════════════════════════════════
── Error (test-functions.R:48:3): body can be a closure ────────────────────────
Error in `function_(d$args, d$body, ..env)`: Body must be an expression or closure.
Backtrace:
    ▆
 1. ├─testthat::expect_equal(fn(x ~ function(y) x + y), foo) at test-functions.R:48:2
 2. │ └─testthat::quasi_label(enquo(object), label, arg = "object")
 3. │   └─rlang::eval_bare(expr, quo_get_env(quo))
 4. └─nofrills::fn(x ~ function(y) x + y)
 5.   └─nofrills function_(d$args, d$body, ..env)

[ FAIL 1 | WARN 0 | SKIP 0 | PASS 107 ]
Error: Test failures
Execution halted

2 errors x | 0 warnings ✓ | 0 notes ✓
````

I don't think it is related to `dplyr`, maybe with `rlang` ?

**Comment by egnha** [comment]
Yes, you're right, this is an rlang-related issue (caused by a change in the behavior of `rlang::is_expression()`). I'll submit a fix to CRAN when rlang 1.0 lands. Thanks for the report.

**Comment by romainfrancois** [comment]
Thanks. cc @lionel-

**Comment by egnha** [comment]
Fixed in 0.3.2.
